### DISCUSSION TITLE
Slider velocity readability change

### DISCUSSION THREAD

**Discussion by MeteMeat** [ppy/osu#30142] [Ideas]
I already posted to reddit about this and was told I should post here, so I'll just copy the post.

I had a minor argument recently with a friend about if slider velocity changes can be sight-read accurately enough. That argument gave me a feature idea. I edited a slider to kind of resemble what I have in mind, but it's pretty hastily made. The ticks are supposed to be about the same length and same distance apart. Keep in mind that this is definitely not how the final implementation would look, this is just what I could quickly make.

![SV_idea](https://github.com/user-attachments/assets/484a94d8-ed0e-4fd5-a43f-beed8e61db57)

These ticks would work kind of like the ticks that are already in game, except their only purpose is to show slider velocity. The denser the ticks, the slower the slider and vice versa. I feel like this would make rapid changes in slider velocity less annoying to deal with.

This would definitely be an optional setting, since i have a feeling some people won't like it and I don't know how it would be implemented in practice since I haven't picked up programming (yet). What do you think about making this a feature?

**Comment by Givikap120** [comment]
I agree with this but, this should be a new mod, similar to mod that makes color of your notes depend on timing
Would be convenient for sightreading maps

However - it's hard to imagine how would you make universally good display for this

Tbh the easiest way of doing this is making sneaking sliders depend on slider velocity instead of AR, but in that case it would indeed be an option instead of mod

**Reply by MeteMeat** [reply]
I'm fine with it being a mod if it gives too much of an advantage. I feel like it could look good in pretty much every context, depends on how dense the ticks need to be with slow sliders. Maybe the ticks could disappear after the slider ball goes over it and / or generate in a way that makes it looks good.

**Reply by Givikap120** [reply]
It doesn't gives too much of an advantage
It gives extra info that you need to learn how to read
For example - similar mod named Synethesia is not used because additional info is hard to convert into something useful
The reason why it should be the mod is because we have precedent for things like this (Synethesia) being a mod.

**Reply by MeteMeat** [reply]
Alright, that makes sense. I think it could maybe be an option at some point, but maybe it could start out as a mod.

**Reply by Givikap120** [reply]
> Alright, that makes sense. I think it could maybe be an option at some point, but maybe it could start out as a mod.

if mod is added - it's not going to be deleted
you need to decide now and stick with it to the end

**Reply by MeteMeat** [reply]
TBH I'd rather it be an option, but is that for me to decide?

**Reply by Givikap120** [reply]
It's up to you to decide how to code it
But the probability of it as an option getting accepted is pretty low
osu!team is very hesitant to add anything new to the settings menu

**Reply by MeteMeat** [reply]
So will I make the code or do I just decide how it should be implemented? Sorry, I'm new to all of this.

**Reply by Givikap120** [reply]
You either make the code or hope that someone will do this for you.
I don't recommend the second variant.

**Reply by MeteMeat** [reply]
I'll try making it myself with some help from friends, but we'll have to wait and see if I'm able to, since I don't have much coding experience (I do have some). I'm sure I can learn though.

**Comment by Fivoka** [comment]
https://github.com/ppy/osu/discussions/13729

I'm bringing this unsearchable discussion post (the OP account is a ghost) here for this conversation.
